plugins/uefi/fu-uefi-device.c

```diff
@@ -289,6 +289,50 @@ fu_uefi_device_build_dp_buf (const gchar *path, gsize *bufsz, GError **error)
 	return g_steal_pointer (&dp_buf);
 }
 
+static GBytes *
+fu_uefi_device_fixup_firmware (FuDevice *device, GBytes *fw, GError **error)
+{
+	FuUefiDevice *self = FU_UEFI_DEVICE (device);
+	g_autoptr(GBytes) fixed_fw = NULL;
+	gsize fw_length;
+	efi_guid_t esrt_guid;
+	efi_guid_t payload_guid;
+	const gchar *data = g_bytes_get_data (fw, &fw_length);
+
+	/* convert to EFI GUIDs */
+	if (efi_str_to_guid (fu_uefi_device_get_guid (self), &esrt_guid) < 0) {
+		g_set_error_literal (error, FWUPD_ERROR, FWUPD_ERROR_INTERNAL, "Invalid ESRT GUID");
+		return NULL;
+	}
+	memcpy (&payload_guid, data, sizeof(efi_guid_t));
+
+	/* ESRT header matches payload */
+	if (efi_guid_cmp (&esrt_guid, &payload_guid) == 0) {
+		g_debug ("ESRT matches payload GUID");
+		fixed_fw = g_bytes_new_from_bytes (fw, 0, fw_length);
+	/* FMP payload */
+	} else if (fu_uefi_device_get_kind (self) == FU_UEFI_DEVICE_KIND_FMP) {
+		g_debug ("performing FMP update");
+		fixed_fw = g_bytes_new_from_bytes (fw, 0, fw_length);
```

Comment:
can we just return with the `g_bytes_new_from_bytes` here?

Reply:
Yep, adjusted.